README.md

```diff
@@ -137,7 +137,152 @@ This now has created the following structure, either on disk or as an in memory
+
+#### new Repo(path[, options])
+
+Creates an IPFS Repo.
+
+Arguments:
+
+* `path` (string, mandatory): the path for this repo
+* `options` (object, optional): may contain the following values
+  * `lock` (string, defaults to `"fs"` in Node.js, `"memory"` in the browser): what type of lock to use. Lock has to be acquired when opening.
+  * `storageBackends` (object, optional): may contain the following values, which should each be a class implementing the [datastore interface](https://github.com/ipfs/interface-datastore#readme):
+    * `root` (defaults to [`datastore-fs`](https://github.com/ipfs/js-datastore-fs#readme) in Node.js and [`datastore-level`](https://github.com/ipfs/js-datastore-level#readme) in the browser)
+    * `blocks` (defaults to [`datastore-fs`](https://github.com/ipfs/js-datastore-fs#readme) in Node.js and [`datastore-level`](https://github.com/ipfs/js-datastore-level#readme) in the browser)
+    * `datastore` (defaults to `datastore-level`)
+
+```js
+const repo = new Repo('path/to/repo')
+```
+
+#### repo.init (callback)
+
+Creates the necessary folder structure inside the repo.
+
+#### repo.open (callback)
+
+Locks the repo.
+
+#### repo.close (callback)
+
+Unlocks the repo.
+
+#### repo.exists (callback)
+
+Tells whether this repo exists or not. Callsback with `(err, bool)`.
+
+### Repos
+
+Root repo:
+
+#### repo.put (key, value:Buffer, callback)
+
+Put a value at the root of the repo.
+
+* `key` can be a buffer, a string or a [Key](https://github.com/ipfs/interface-datastore#keys).
+
+#### repo.get (key, callback)
+
+Get a value at the root of the repo.
+
+* `key` can be a buffer, a string or a [Key](https://github.com/ipfs/interface-datastore#keys).
+* `callback` is a callback function `function (err, result:Buffer)`
+
+[Blocks](https://github.com/ipfs/js-ipfs-block#readme):
+
+#### repo.blocks.put (block:Block, callback)
+
+* `block` should be of type [Block](https://github.com/ipfs/js-ipfs-block#readme).
+
+#### repo.blocks.putMany (blocks, callback)
+
+Put many blocks.
+
+* `block` should be an array of type [Block](https://github.com/ipfs/js-ipfs-block#readme).
+
+#### repo.blocks.get (cid, callback)
+
+Get block.
+
+* `cid` is the content id of [type CID](https://github.com/ipld/js-cid#readme).
+* `callback` is a callback function `function (err, result:Buffer)`
+
+Datastore:
+
+#### repo.datastore
+
+This is contains a full implementation of [the `interface-datastore` API](https://github.com/ipfs/interface-datastore#api).
+
+
+### Utils
+
+#### repo.config
+
+Instead of using `repo.set('config')` this exposes an API that allows you to set and get a decoded config object, as well as, in a safe manner, change any of the config values individually.
+
+##### repo.config.set(key:string, value, callback)
+
+Set a config value. `value` can be any object that is serializable to JSON.
+
+* `key` is a string specifying the object path. Example:
+
+```js
+repo.config.set('a.b.c', 'c value', (err) => {
+  if (err) throw err
```

Comment:
please use `{}`, even if they are all in the same line so that it follows our agreed codestyle